src/Test/Test.jl

```diff
@@ -36,6 +36,10 @@ struct GenericGraph{T} <: Graphs.AbstractGraph{T}
     g::SimpleGraph{T}
 end
 
+function GenericGraph(elist::Vector{Graphs.SimpleGraphEdge{T}}) where {T<:Integer}
+    GenericGraph{T}(SimpleGraph(elist))
+end
```

Comment:
@thchr I spotted this too late - I think we should not implement any method (this includes constructors) that specialize on a `GenericGraph` or `GenericDiGraph`. The reason is, that we want to use `GenericGraph` as a placeholder for an arbitrary `AbstractGraph` to verify functions that take an `AbstractGraph` work correctly.

Reply:
I mean the issue I see here is, if someone writes a function like
```julia
function f(g::AbbstractGraph)
    T = typeof(g)
    return T([Edge(1, 2)])
end
```
then this function might fail for an arbitrary `AstractGraph` but not for a `GenericGraph`. I don't think it will break anything soon though, so there is no rush, but I can make a PR that works around that.

I would propose, if we need a workaround for constructors so that it will be easier to write tests, we just implement the functions `generic_graph` and `generic_digraph` -  maybe I already did such a thing in one of my PRs but I don't remember exactly.

Reply:
Sounds good to me :)